salt/utils/gitfs.py

```diff
@@ -1599,9 +1599,15 @@ def _traverse(tree, blobs, prefix):
             _traverse(tree, blobs, self.root(tgt_env))
         add_mountpoint = lambda path: salt.utils.path_join(self.mountpoint(tgt_env), path)
         for repo_path in blobs.get('files', []):
-            files.add(add_mountpoint(relpath(repo_path)))
+            if not salt.utils.is_windows():
+                files.add(add_mountpoint(relpath(repo_path)))
+            else:
+                files.add('/'.join(repo_path.replace('.:\\', '', 1).split(os.sep)))
```

Comment:
Yeah, for all I know this is a problem in pygit2, but if it's something that I can fix by handling the traversal better, then we don't have to resort to the platform specific fix you have here.